iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```
@@ -349,6 +365,96 @@ public Map<String, String> getBasicStatistics(Partish partish) {
+        String statsPath = table.location() + STATS + snapshotId;
+        LOG.info("Using stats from puffin file at:" + statsPath);
+        try (PuffinReader reader = Puffin.read(table.io().newInputFile(statsPath)).build()) {
+          BlobMetadata blobMetadata = reader.fileMetadata().blobs().get(0);
+          Map<BlobMetadata, List<ColumnStatistics>> collect =
+              Streams.stream(reader.readAll(ImmutableList.of(blobMetadata))).collect(Collectors.toMap(Pair::first,
+                  blobMetadataByteBufferPair -> SerializationUtils.deserialize(
+                      ByteBuffers.toByteArray(blobMetadataByteBufferPair.second()))));
+
+          return collect.entrySet().stream().iterator().next().getValue().get(0).getStatsObj();
+        } catch (IOException e) {
+          LOG.info(String.valueOf(e));
+        }
+        break;
+      default:
+        // fall back to metastore
+    }
+    return null;
+  }
+
+
+  @Override
+  public boolean setColStatistics(org.apache.hadoop.hive.ql.metadata.Table table,
+      List<ColumnStatistics> colStats) {
+    TableDesc tableDesc = Utilities.getTableDesc(table);
+    Table tbl = Catalogs.loadTable(conf, tableDesc.getProperties());
+    String snapshotId = tbl.name() + tbl.currentSnapshot().snapshotId();
+    byte[] serializeColStats = SerializationUtils.serialize((Serializable) colStats);
+
+    try (PuffinWriter writer = Puffin.write(tbl.io().newOutputFile(tbl.location() + STATS + snapshotId))
+        .createdBy("Hive").build()) {
+      writer.add(
+          new Blob(
+              tbl.name() + "-" + snapshotId,
+              ImmutableList.of(1),
+              tbl.currentSnapshot().snapshotId(),
+              tbl.currentSnapshot().sequenceNumber(),
+              ByteBuffer.wrap(serializeColStats),
+              PuffinCompressionCodec.NONE,
+              ImmutableMap.of()));
+      writer.finish();
+    } catch (IOException e) {
+      LOG.info(String.valueOf(e));
```

do not swallow exception